fedimint-client/src/lib.rs

```diff
@@ -678,18 +675,12 @@ impl ClientHandle {
         self.shutdown_inner().await;
     }
 
+    /// Shutdown the client. `self` MUST be dropped after this call.
```

Comment:
Note for reviewer: The API contract documented here was already applicable before this PR, since it was expected that `self.inner` was always `Some` (hence all of the `expect()`s we're removing) except in the `Drop` impl of `ClientHandle`